Disable two libgit2 tests with libgit2 < 0.26.0

nalimilan opened the pull request:
Without our patches libgit2-remote-push-NULL and libgit2-gitconfig-symlink,
which are not yet included in an upstream release, these tests fail.
Disable them so that builds using the system libgit2 can pass the tests.

See https://github.com/JuliaLang/julia/pull/21597#issuecomment-300909309, https://github.com/JuliaLang/julia/issues/21639 and #21636.

nalimilan committed:
Disable two libgit2 tests with libgit2 < 0.26.0

Without our patches libgit2-remote-push-NULL and libgit2-gitconfig-symlink,
which are not yet included in an upstream release, these tests fail.
Disable them so that builds using the system libgit2 can pass the tests.

nalimilan commented:
I don't know, but if we ask them to do they would probably accept given that the patch is simple and cannot break existing code. Or to be safe we could check that libgit2 >= 0.26.

tkelman commented:
I just asked on their slack channel, will see what they say. Looking at the response in https://github.com/libgit2/libgit2/issues/4220 it sounds like they might not want to do an 0.25.2 and want to do 0.26.0 soon, but if that has API changes it would require more work on our side.

How hard is it to get trivial patches into the fedora libgit2 package?

nalimilan commented:
> I just asked on their slack channel, will see what they say. Looking at the response in libgit2/libgit2#4220 it sounds like they might not want to do an 0.25.2 and instead plan on doing 0.26.0 soon, but if that has API changes it would require more work on our side.

Anyway we'll use 0.26 at some point, won't we? There's no real hurry to enable this test, which wasn't present in 0.5.

> How hard is it to get trivial patches into the fedora libgit2 package?

Not really hard, but I'd rather have Julia build with the system libgit2 than depend on custom patches. There are other distribution packages out there, and people may want to use their system libgit2 for some reason.

tkelman commented:
Fair enough. Let's enable the test for `>= v"0.26.0"` then, unless I hear back soon about whether there will be an 0.25.2

nalimilan commented:
OK, I've updated the patch. I've also realized that after disabling this test, another test from https://github.com/JuliaLang/julia/pull/21636 failed for the same reasons, so I've added a condition for it too. With this, all tests pass when building the Fedora RPM.
